controllers/scopetemplate_controller_test.go

```diff
@@ -0,0 +1,548 @@
+					UID:        scopeTemplate.GetUID(),
+				}))
+				Expect(existingRole.Rules).Should(Equal([]rbacv1.PolicyRule{
+					{
+						Verbs:     []string{"get", "watch", "list"},
+						APIGroups: []string{""},
+						Resources: []string{"secrets"},
+					},
+				}))
+			})
+
+			It("creates the expected RoleBinding for test namespace", func() {
+				roleBindingList := &rbacv1.RoleBindingList{}
+				Eventually(func() error {
+					err := k8sClient.List(ctx, roleBindingList, &client.ListOptions{
+						Namespace: namespace.GetName(),
+					},
+						client.MatchingLabels{
+							scopeInstanceUIDKey:           string(scopeInstance.GetUID()),
+							scopeTemplateUIDKey:           string(scopeTemplate.GetUID()),
+							clusterRoleBindingGenerateKey: "test",
+						})
+					if err != nil {
+						return err
+					}
+
+					if len(roleBindingList.Items) != 1 {
+						return fmt.Errorf("Expected 1 roleBinding, found %d", len(roleBindingList.Items))
+					}
+
+					return nil
+				}, timeout, interval).Should(BeNil())
+
+				existingRB := &roleBindingList.Items[0]
+
+				Expect(len(existingRB.OwnerReferences)).To(Equal(1))
+				Expect(existingRB.OwnerReferences).Should(ContainElement(metav1.OwnerReference{
+					APIVersion: "operators.io.operator-framework/v1",
+					Kind:       "ScopeInstance",
+					Name:       scopeInstance.GetObjectMeta().GetName(),
+					UID:        scopeInstance.GetObjectMeta().GetUID(),
+				}))
+
+				Expect(len(existingRB.Subjects)).To(Equal(1))
+				Expect(existingRB.Subjects).Should(ContainElement(rbacv1.Subject{
+					Kind:     "Group",
+					Name:     "manager",
+					APIGroup: "rbac.authorization.k8s.io",
+				}))
+				Expect(existingRB.RoleRef).To(Equal(rbacv1.RoleRef{
+					Kind:     "ClusterRole",
+					Name:     "test",
+					APIGroup: "rbac.authorization.k8s.io",
+				}))
+			})
+
+			When("a scopeInstance is updated to include another namespace", func() {
+
+				BeforeEach(func() {
+					namespace2 = &corev1.Namespace{
+						ObjectMeta: metav1.ObjectMeta{
+							GenerateName: "test-",
+						},
+					}
+					Expect(k8sClient.Create(ctx, namespace2)).NotTo(HaveOccurred())
+				})
+
+				AfterEach(func() {
+					Expect(k8sClient.Delete(ctx, namespace2)).NotTo(HaveOccurred())
+				})
+
+				It("should not create a clusterRole", func() {
+					Eventually(func() error {
+						// TODO: scopeTemplate should check for status that no clusterRole exists.
+						// Until then, just make sure no clusterRole exists
+						clusterRoleList := &rbacv1.ClusterRoleList{}
+						if err := k8sClient.List(ctx, clusterRoleList, client.MatchingLabels{
+							scopeTemplateUIDKey:    string(scopeTemplate.GetUID()),
+							clusterRoleGenerateKey: "test",
+						}); err != nil {
+							return err
+						}
+
+						if len(clusterRoleList.Items) != 0 {
+							return fmt.Errorf("Expected 0 clusterRoles, found %d", len(clusterRoleList.Items))
+						}
+
+						return nil
+
+					}, timeout, interval).Should(BeNil())
+				})
```

Comment:
This could also be moved to its own function, parameters should include the MatchingLabels and the number of expected roles.

Reply:
Updated.